### PR TITLE
support tasks

### PR DESCRIPTION
**Pull request by Ed-XCF**

### CODE REVIEW

**Comment by JoshYuJump** [comment]
Are there any usage examples? @Ed-XCF

**Comment by Ed-XCF** [comment]
How to use:
-----main.py or web.py-----
from bali.tasks import add_background_tasks
from fastapi import FastAPI

add_background_tasks(app)

-----tasks.py-----
from bali.tasks import run_every

@run_every(30)
def check_half_payment_status():
    pass

File: bali/tasks.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                            loop = asyncio.get_event_loop()
+                            await loop.run_in_executor(None, func)
+                    except Exception as e:
+                        logger.error("Task %s raise an error: %s", task_name, repr(e))
+                    else:
+                        logger.info("Task %s done", task_name)
+                    finally:
+                        await asyncio.sleep(seconds)
+
+            asyncio.ensure_future(task())
```
Comment (JoshYuJump) [inline bali/tasks.py:46]: I wonder if you can replace `ensure_future` with `create_task` here. 

Bali required Python >= 3.7.

`ensure_future` it's a low-level function and will marked as deprecated in Python 3.10

**Comment by JoshYuJump** [comment]
Is it work well when the app running in multi workers or multi pods in K8S?

**Comment by Ed-XCF** [comment]
> Is it work well when the app running in multi workers or multi pods in K8S?

There is an idea to reduce the possibility of synchronous startup of tasks through mathematical statistics, but users must be concerned about concurrency and locking

**Comment by JoshYuJump** [comment]
I'll closed it as there's no update for a long time for this PR.

**Pull request closed without merging**